Mockito dependency changed + GreetingController test made with Mockito Extension to JUnit

BrainDias opened the pull request:
Mockito dependency changed to testImplementation
+
GreetingController test made with annotations

BrainDias committed:
Mockito dependency type replaced with testImplementation
+
Injected the mock of the service in GreetingControllerTests unit test with annotation

ahlinist commented:
PR title should reflect the entire body of work: you not only changed the dependency scope, but also modified the test itself.

ahlinist commented:
It would be better to set assignee and reviewers explicitly .

ahlinist commented on src/test/java/com/example/demo/controller/GreetingControllerTests.java:
```diff
@@ -9,11 +10,12 @@
 
 class GreetingControllerTests {
 
+@InjectMocks
```
Wrong indentation, please add 4 white-spaces.

ahlinist commented on src/test/java/com/example/demo/controller/GreetingControllerTests.java:
```diff
@@ -9,11 +10,12 @@
 class GreetingControllerTests {
 
+@InjectMocks
+GreetingService serviceMock;
```
The mocks are supposed to be injected into a class that is under test. Not the one that is supposed to be mocked. This filed should be changed to: `private GreetingController greetingController;`

ahlinist commented on src/test/java/com/example/demo/controller/GreetingControllerTests.java:
```diff
@@ -9,11 +10,12 @@
-		Mockito.when(ServiceMock.printHello(name)).thenReturn("Hello "+name+"!");
-		assertEquals(new GreetingController(ServiceMock).hello(name), "Hello " + name + "!");
+		Mockito.when(serviceMock.printHello(name)).thenReturn("Hello "+name+"!");
+		assertEquals(new GreetingController(serviceMock).hello(name), "Hello " + name + "!");
```
Mocked service is supposed to be defined as a class-member field somehow like this:

```
@Mock
private GreetingService greetingService;
```

BrainDias committed:
Mocking fields by annotations with MockitoExtension for test class

ahlinist commented on src/test/java/com/example/demo/controller/GreetingControllerTests.java:
```diff
@@ -1,19 +1,25 @@
-		GreetingService ServiceMock = Mockito.mock(GreetingService.class);
-		Mockito.when(ServiceMock.printHello(name)).thenReturn("Hello "+name+"!");
-		assertEquals(new GreetingController(ServiceMock).hello(name), "Hello " + name + "!");
+		Mockito.when(serviceMock.printHello(name)).thenReturn("Hello " + name + "!");
```
Use static import for when. And the whole string for return value.

ahlinist commented on src/test/java/com/example/demo/controller/GreetingControllerTests.java:
```diff
@@ -1,19 +1,25 @@
-		Mockito.when(ServiceMock.printHello(name)).thenReturn("Hello "+name+"!");
-		assertEquals(new GreetingController(ServiceMock).hello(name), "Hello " + name + "!");
+		Mockito.when(serviceMock.printHello(name)).thenReturn("Hello " + name + "!");
+		assertEquals(greetingControllerWithMocks.hello(name), "Hello " + name + "!");
```
Compare expected result with plain string. Not constructed one.

BrainDias committed:
1. Changed field names in GreetingControllerTests
2. Replaced generated string with plain string.
3. Created Implementation for GreetingService

BrainDias committed:
Deleted interface for service

BrainDias committed:
Only changed files left
